src/explore-education-statistics-admin/src/pages/release/data/components/__tests__/DataFilesTableRow.test.tsx

```diff
@@ -0,0 +1,106 @@
+﻿import React from 'react';
+import { MemoryRouter } from 'react-router-dom';
+import { fireEvent, render, screen, waitFor } from '@testing-library/react';
+import { FeatureFlags } from '@admin/config/featureFlags';
+import { FeatureFlagProvider } from '@admin/contexts/FeatureFlagContext';
+import DataFilesTableRow from '../DataFilesTableRow';
+
+describe('DataFilesTableRow', () => {
+  const baseProps = {
+    canUpdateRelease: true,
+    publicationId: 'pub-1',
+    releaseVersionId: 'release-1',
+    onConfirmDelete: jest.fn(),
+    onStatusChange: jest.fn(),
+  };
+
+  const mockDataFile = {
+    fileName: '',
+    metaFileName: '',
+    metaFileId: '',
+    userName: '',
+    id: 'file-1',
+    title: 'Test File',
+    status: 'COMPLETE' as const,
+    fileSize: { size: 1000, unit: 'B' },
+    permissions: { canCancelImport: false },
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('when feature flag is toggled', () => {
+    test('shows no modal when feature flag is enabled', async () => {
```

Comment:
Probably worth mentioning the feature flag name in the title, as `FAIL when feature flag is toggled => shows no modal when feature flag is enabled` isn't very descriptive